File: python/src/common/calendar.py
import datetime
import json
from common.file import CommonFile

class CommonCalendar():
    def __init__(self):
        # 祝日情報を読み込み
        self.load_holidays()
        pass
    
    holidays :dict = {}
    
    @classmethod
    def load_holidays(cls):
        if cls.holidays:
            return cls.holidays
        target_file_path = 'src/common/jsons/holidays.json'
        file_path = CommonFile.create_abs_path(target_file_path)
        
        file_data = CommonFile.read(file_path)
        json_data = json.loads(file_data)
        cls.holidays = json_data
        return cls.holidays
    
    @classmethod
    def is_holiday(cls, date=None):
        cls.load_holidays()
        
        # 未指定時は今日で判定
        if not date:
            date = datetime.date.today()

        # 土日判定
        if date.weekday() in [5, 6]:
            return True

        # 祝日判定
        str_date = date.strftime('%Y-%m-%d')
        if str_date in cls.holidays:
            return True
        
        return False
        
        

File: csharp/SeleniumSample/Src/Common/Calendar.cs
using System.Text.Json;

namespace Common
{
    class CommonCalendar
    {
        public static Dictionary<string, string> holidays = new Dictionary<string, string>() { };

        public static Dictionary<string, string> LoadHolidays()
        {
            if (CommonCalendar.holidays.Count != 0)
            {
                return CommonCalendar.holidays;
            }
            var targetFilePath = "common/jsons/holidays.json";
            var filePath = CommonFile.CreateAbsPath(targetFilePath);
            var fileData = CommonFile.Read(filePath);
            var jsonData = JsonSerializer.Deserialize(fileData, typeof(Dictionary<string, string>));

            CommonCalendar.holidays = (Dictionary<string, string>)(jsonData ?? new Dictionary<string, string> { });
            return CommonCalendar.holidays;
        }

        public static bool IsHoliday(DateTime? argDate = null)
        {

            // 未指定時は今日で判定
            DateTime date;
            if (argDate == null)
            {
                date = DateTime.Now;
            }
            else
            {
                date = argDate.Value;
            }

            // 土日判定
            var demo = new List<DayOfWeek>() { DayOfWeek.Saturday, DayOfWeek.Sunday };
            if (demo.Contains(date.DayOfWeek))
            {
                return true;
            }

            // 祝日判定
            var strDate = date.ToString("yyyy-MM-dd");
            if (CommonCalendar.LoadHolidays().ContainsKey(strDate))
            {
                return true;
            }

            return false;
        }

    }
}